VideoLocker/src/main/java/org/edx/mobile/base/BaseFragmentActivity.java

```diff
@@ -340,6 +414,30 @@ public void stopAnimation(View view) {
     }
 
     /**
+     * Display the banner at the top of the Activity.
+     *
+     * @param bannerType The banner type
+     */
+    @Override
+    public void showBanner(@NonNull BannerType bannerType) {
+        bannerTextView.setText(bannerType.getShortMessageRes(this));
+        bannerExpanded.setText(bannerType.getLongMessageRes(this));
+        bannerExpanded.setOnClickListener(bannerType.getClickListener());
+        banner.setVisibility(View.VISIBLE);
+        displayedBannerType = bannerType;
+    }
+
+    /**
+     * Hide the banner at the top of the Activity.
+     */
+    private void hideOfflineBanner() {
```

Comment:
The banner display is mostly handled by `BaseFragmentActivity` itself, and doesn't need to be performed from subclasses. The `showBanner()` method is public because it's the implementation of the `BannerDisplayCallback` interface, which is implemented here so that it could be called from `ErrorUtils` by the `Task` implementation. The upgrade header should never need to be hidden unless the app is actually upgraded, while hiding the offline header at the appropriate time is handled by `BaseFragmentActivity` itself.